posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -110,6 +118,13 @@ def _evolve_pyarrow_schema(table: pa.Table, delta_schema: deltalake.Schema | Non
             json_column = pa.array([json.dumps(row.as_py()) if row.as_py() is not None else None for row in column])
             table = table.set_column(table.schema.get_field_index(column_name), column_name, json_column)
             column = table.column(column_name)
+        # Change pa.duration to int with total seconds
+        elif pa.types.is_duration(column.type):
+            seconds_column = pa.array(
+                [row.as_py().total_seconds() if row.as_py() is not None else None for row in column]
+            )
+            table = table.set_column(table.schema.get_field_index(column_name), column_name, seconds_column)
+            column = table.column(column_name)
```

Comment:
nit: Every `row.as_py()` call allocates memory for the python type (a `timedelta`, I assume in this case). We could save some memory here by using pyarrow operations:

```suggestion
            if column.unit == "s":
                factor = 1
            elif column.unit == "ms":
                factor = 1_000
            elif column.unit == "us":
                factor = 1_000_000
            elif column.unit == "ns":
                factor = 1_000_000_000
            else:
                # Should never get here as we have covered all possible units.
                # But there is no way to assert this at compile time.
                # See for possible units: https://arrow.apache.org/docs/python/generated/pyarrow.duration.html
                raise ValueError(f"Invalid unit: {column.unit}")

            # This gets us an Int64 array, which is the same as what we would get
            # by creating an array using Python's int.
            second_column = pc.multiply(column.cast(pa.int64()), factor)
            table = table.set_column(table.schema.get_field_index(column_name), column_name, seconds_column)
            column = table.column(column_name)
```

Reply:
Great knowledge - thank you